File: pkg/day05/day05_test.go
package day05

import (
	"testing"
)

func TestDay05Part1Example1(t *testing.T) {
	input, err := load("testdata/example1.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 143
	got := input.part1()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestDay05Part1(t *testing.T) {
	input, err := load("testdata/input.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 6498
	got := input.part1()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestDay05Part2Example1(t *testing.T) {
	input, err := load("testdata/example1.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 123
	got := input.part2()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestDay05Part2(t *testing.T) {
	input, err := load("testdata/input.txt")
	if err != nil {
		t.Fatal(err)
	}

	want := 5017
	got := input.part2()

	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

File: pkg/day05/day05_test.rb
require "minitest/autorun"
require "minitest/focus"
require "minitest/reporters"

Minitest::Reporters.use!

require_relative "day05"

class Day05Test < Minitest::Test
  def test_part1_example1
    input = Day05.new("testdata/example1.txt")
    assert_equal(0, input.part1)
  end

  def test_part1
    input = Day05.new("testdata/input.txt")
    assert_equal(0, input.part1)
  end

  def test_part2_example1
    input = Day05.new("testdata/example1.txt")
    assert_equal(0, input.part2)
  end

  def test_part2
    input = Day05.new("testdata/input.txt")
    assert_equal(0, input.part2)
  end
end
